hassio/src/ingress-view/hassio-ingress-view.ts

```diff
@@ -44,18 +47,106 @@ class HassioIngressView extends LitElement {
 
   private _sessionKeepAlive?: number;
 
+  private _resolveIngressURL: {
+    status?: number;
+    time?: number;
+    interval?: number;
+  } = {};
+
   public disconnectedCallback() {
     super.disconnectedCallback();
 
     if (this._sessionKeepAlive) {
       clearInterval(this._sessionKeepAlive);
       this._sessionKeepAlive = undefined;
     }
+    if (this._resolveIngressURL.interval) {
+      clearInterval(this._resolveIngressURL.interval);
+    }
+    this._resolveIngressURL = {};
   }
 
-  protected render(): TemplateResult {
+  public connectedCallback() {
+    super.connectedCallback();
+
+    this._resolveURL();
+    this._resolveIngressURL.interval = window.setInterval(async () => {
+      await this._resolveURL();
+    }, 1000);
+  }
+
+  private async _resolveURL(): Promise<void> {
     if (!this._addon) {
-      return html` <hass-loading-screen></hass-loading-screen> `;
+      return;
+    }
+
+    if (this._addon.state !== "started") {
+      clearInterval(this._resolveIngressURL.interval);
+      await showAlertDialog(this, {
+        text:
+          this.hass.localize("ingress.not_running") ||
+          "The add-on is not running, please start it.",
+        title: this._addon.name,
+        confirmText:
+          this.hass.localize("ingress.go_to_dashboard") ||
+          "Go to add-on dashboard",
+      });
+      await nextRender();
+      navigate(`/hassio/addon/${this._addon.slug}/info`, { replace: true });
+      return;
+    }
+
+    if (
+      this._resolveIngressURL.status &&
+      this._resolveIngressURL.status !== STATUS_BAD_GATEWAY
+    ) {
+      if (this._resolveIngressURL.interval) {
+        clearInterval(this._resolveIngressURL.interval);
+      }
+    }
+
+    if (
+      this._resolveIngressURL.time &&
```

Comment:
Where is `time` set?

Reply:
Nowhere apparently... 
I remember testing it, so it was there at some point